src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -734,6 +736,27 @@ function registerRazorCommands(context: vscode.ExtensionContext, languageServer:
     );
 }
 
+function registerSourceGeneratedFilesContentProvider(
+    context: vscode.ExtensionContext,
+    languageServer: RoslynLanguageServer
+) {
+    context.subscriptions.push(
+        vscode.workspace.registerTextDocumentContentProvider(
+            'source-generated',
+            new (class implements vscode.TextDocumentContentProvider {
+                async provideTextDocumentContent(uri: vscode.Uri, token: vscode.CancellationToken): Promise<string> {
+                    const result = await languageServer.sendRequest(
+                        RoslynProtocol.SourceGeneratorGetTextRequest.type,
+                        { textDocument: { uri: UriConverter.serialize(uri) } },
+                        token
+                    );
+                    return result.text;
+                }
+            })()
```

Comment:
so i personally really dont' like these sorts of anonymous inner classes.  my pref is a distinct class with clear cpatures.  but i don't feel strongly enough to block.  if you feel like this is ok, then i'm ok with it.